### PR TITLE
LibJS: Fix some memory tracking issues

### PR DESCRIPTION
**Pull request by davidot**
`test-js -g` now works again! :tada:
Just for reference it takes about ~100 times as much time as the normal `test-js` on serenity.
So although it is a good test we probably still don't want it in CI.

Also remove crashing for modules in the REPL and remove the somewhat bitrotted JS_TRACK_ZOMBIE_CELLS option.

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/VM.cpp [hunk 1]
```diff
@@ -989,7 +989,7 @@ void VM::finish_dynamic_import(ScriptOrModule referencing_script_or_module, Modu
     dbgln_if(JS_MODULE_DEBUG, "[JS MODULE] finish_dynamic_import on {}", module_request.module_specifier);
 
     // 1. Let fulfilledClosure be a new Abstract Closure with parameters (result) that captures referencingScriptOrModule, specifier, and promiseCapability and performs the following steps when called:
-    auto fulfilled_closure = [referencing_script_or_module, module_request = move(module_request), promise_capability](VM& vm, GlobalObject& global_object) -> ThrowCompletionOr<Value> {
+    auto fulfilled_closure = [referencing_script_or_module, module_request = move(module_request), resolve_function = make_handle(promise_capability.resolve), reject_function = make_handle(promise_capability.reject)](VM& vm, GlobalObject& global_object) -> ThrowCompletionOr<Value> {
```
Comment (awesomekling) [inline Userland/Libraries/LibJS/Runtime/VM.cpp:991]: Is this actually needed? `promise_capability` is on the stack, and that should be visible to the conservative GC scan.

File: Userland/Libraries/LibJS/Runtime/PromiseResolvingElementFunctions.cpp [hunk 2]
```diff
@@ -13,6 +13,12 @@
 
 namespace JS {
 
+void PromiseValueList::visit_edges(Visitor& visitor)
+{
```
Comment (awesomekling) [inline Userland/Libraries/LibJS/Runtime/PromiseResolvingElementFunctions.cpp:17]: Please call up to the base class `visit_edges()` here as well :)

File: Userland/Libraries/LibJS/Runtime/PromiseResolvingElementFunctions.h [hunk 3]
```diff
@@ -35,6 +35,7 @@ class PromiseValueList final : public Cell {
 
 private:
     virtual const char* class_name() const override { return "PromiseValueList"; }
+    virtual void visit_edges(Visitor& visitor) override;
```
Comment (awesomekling) [inline Userland/Libraries/LibJS/Runtime/PromiseResolvingElementFunctions.h:38]: Please omit parameter names that are exactly the same as the type name.

File: Userland/Libraries/LibJS/Runtime/VM.cpp (hunk 1 above)
Comment (davidot) [reply Userland/Libraries/LibJS/Runtime/VM.cpp:991]: It is on the stack here, but these function might only be called later when we are running queued promises. In that case these could be the only references to the promise capability values.

To make it concrete without this `test-js -g` fails.

Comment (awesomekling) [reply Userland/Libraries/LibJS/Runtime/VM.cpp:991]: Ah, I see! :)

**Pull request merged**